### ISSUE TITLE
grpc api and remote client

### ISSUE THREAD

**Issue by nics** [ugent-library/biblio-backend#485]
- [ ] get publication
- [ ] stream all publications
- [ ] update publication
- [ ] import publications
- [ ] search publications
- [ ] upload file
- [ ] download file
- [ ] get dataset
- [ ] stream all datasets
- [ ] update dataset
- [ ] import datasets
- [ ] search datasets
- [ ] grpc zap logging
- [ ] grpc authentication

**Comment by netsensei** [comment]
Added: TLS 1.3 support for encrypting communication between client / server.
See: https://github.com/ugent-library/biblio-backend/wiki/Setup-TLS

**Comment by mietcls** [comment]
Leftover check boxes are for dogfooding purposes

**Comment by netsensei** [comment]
I've added new to-be-ported commands which have been implemented in the legacy CLI app over the course of the past few months to the list. We need these in the gRPC client as well.

**Comment by netsensei** [comment]
Gave the ticket an update with acceptance criteria & nice to have's

**Comment by mietcls** [comment]
Checked off steps we did and the ones we will not do

**Comment by netsensei** [comment]
@mietcls I've cleaned this issue per our conversation yesterday.  The minimum we need to do before we can close this:

* Add tests for adding & importing files.
* Write technical and usage documentation.

Reviewing the functionality of the command, it covers everything we, the team, need to remote administer the backoffice. The gRPC service has arrived at a point where it can be used to consider porting the bash and other scripts @nics uses to manage / munge data.

Once we've done the left-over todo's, we can close this and discuss whatever else we need in terms of functionality or improvement in separate issues. (e.g. #829)

**Comment by mietcls** [comment]
As we are not going to build a CLI client that will be used outside of our team any time soon; we will not prioritise this now.

**Comment by mietcls** [comment]
I've been told to close this
